tests/test_probes.py

```diff
@@ -377,28 +385,53 @@ class Entity(object):
             # The next problem is that, while our patch is live,
             # if t2 goes out of its original scope, we've still got
             # a reference to it in our mock patch.
-            self.assertEqual(
-                owner_types(func_2),
-                {
+            if six.PY2:
+                expected_result = {
                     types.ModuleType: 2,
                     Entity: 2,
                     probes.WeakMethodPatch: 3,
                     MockPatch: 1,
                     probes.DictPatch: 1,
-                },
-            )
-            del t2
+                }
+            else:
+                expected_result =  defaultdict(int, {
+                      types.FunctionType: 1,
+                      types.ModuleType: 2,
+                      MockPatch: 1,
+                      diagnose.probes.WeakMethodPatch: 4,
+                      diagnose.probes.DictPatch: 1,
+                      Entity: 2
+                })
```

Comment:
I don't like this defaultdict at first glance. If Python 3 is making more references to an object than Py2, I want to know exactly what those are. I'll check out this branch and do some digging.